rqt_py_common/src/rqt_py_common/rosaction.py

```diff
@@ -1,884 +0,0 @@
-        if '/' in arg:  # package specified
-            rosaction_debug(rospack, mode, arg, options.raw)
-        else:
-            for found in rosaction_search(rospack, mode, arg):
-                print("[%s]:" % found)
-                rosaction_debug(rospack, mode, found, options.raw)
-
-
-def rosaction_md5(mode, type_):
-    try:
-        if mode == MODE_ACTION:
-            msg_class = roslib.message.get_message_class(type_)
-        else:
-            msg_class = roslib.message.get_service_class(type_)
-    except ImportError:
-        raise IOError("cannot load [%s]" % (type_))
-    if msg_class is not None:
-        return msg_class._md5sum
-    else:
-        raise IOError("cannot load [%s]" % (type_))
-
-
-def rosaction_cmd_md5(mode, full):
-    parser = OptionParser(usage="usage: ros%s md5 <%s>" % (mode[1:], full))
-    options, arg = _stdin_arg(parser, full)
-
-    if '/' in arg:  # package specified
-        try:
-            md5 = rosaction_md5(mode, arg)
-            print(md5)
-        except IOError:
-            print("Cannot locate [%s]" % arg, file=sys.stderr)
-    else:
-        rospack = rospkg.RosPack()
-        matches = [m for m in rosaction_search(rospack, mode, arg)]
-        for found in matches:
-            try:
-                md5 = rosaction_md5(mode, found)
-                print("[%s]: %s" % (found, md5))
-            except IOError:
-                print("Cannot locate [%s]" % found, file=sys.stderr)
-        if not matches:
-            print("No messages matching the name [%s]" % arg, file=sys.stderr)
-
-
-def rosaction_cmd_package(mode, full):
-    parser = OptionParser(usage="usage: ros%s package <package>" % mode[1:])
-    parser.add_option("-s",
-                      dest="single_line", default=False, action="store_true",
-                      help="list all msgs on a single line")
-    options, arg = _stdin_arg(parser, full)
-    joinstring = '\n'
-    if options.single_line:
-        joinstring = ' '
-    print(joinstring.join(list_types(arg, mode=mode)))
-
-
-def rosaction_cmd_packages(mode, full, argv=None):
-    if argv is None:
-        argv = sys.argv[1:]
-    parser = OptionParser(usage="usage: ros%s packages" % mode[1:])
-    parser.add_option("-s",
-                      dest="single_line", default=False, action="store_true",
-                      help="list all packages on a single line")
-    options, args = parser.parse_args(argv[1:])
-    rospack = rospkg.RosPack()
-    joinstring = '\n'
-    if options.single_line:
-        joinstring = ' '
-    p1 = [p for p, _ in iterate_packages(rospack, mode)]
-    p1.sort()
-    print(joinstring.join(p1))
-
-
-def rosaction_debug(rospack, mode, type_, raw=False):
-    """
-    Prints contents of msg/srv file
-    :param mode: MODE_ACTION or MODE_SRV, ``str``
-    """
-    if mode == MODE_ACTION:
-        print(get_msg_text(type_, raw=raw, rospack=rospack))
-    else:
-        raise ROSActionException("Invalid mode for debug: %s" % mode)
```

Comment:
Removing this file and adding it later again will "loose" the history and is very unfortunate.

Reply:
According to the todo's this file needs to be moved to actionlib. I didn't know it was possible to move a file across repos and preserve history